app/src/org/commcare/activities/DispatchActivity.java

```diff
@@ -321,14 +330,24 @@ private void handleUnvalidatedApp() {
     private void handleExternalLaunch() {
         //First off, make sure the incoming session is clear
         CommCareApplication.instance().getSession().proceedWithSavedSessionIfNeeded(() -> {
-                    String sessionRequest = this.getIntent().getStringExtra(SESSION_REQUEST);
-                    SessionStateDescriptor ssd = new SessionStateDescriptor();
-                    ssd.fromBundle(sessionRequest);
-                    CommCareApplication.instance().getCurrentSessionWrapper().loadFromStateDescription(ssd);
-                    Intent i = new Intent(this, StandardHomeActivity.class);
-                    i.putExtra(WAS_EXTERNAL, true);
-                    startActivityForResult(i, HOME_SCREEN);
-        }
+                    if (getIntent().hasExtra(SESSION_REQUEST)) {
+                        String sessionRequest = this.getIntent().getStringExtra(SESSION_REQUEST);
+                        SessionStateDescriptor ssd = new SessionStateDescriptor();
+                        ssd.fromBundle(sessionRequest);
+                        CommCareApplication.instance().getCurrentSessionWrapper().loadFromStateDescription(ssd);
+                        Intent i = new Intent(this, StandardHomeActivity.class);
+                        i.putExtra(WAS_EXTERNAL, true);
+                        startActivityForResult(i, HOME_SCREEN);
+                    } else if (getIntent().hasExtra(SESSION_ENDPOINT_ID)) {
+                        String sessionEndpointId = this.getIntent().getStringExtra(SESSION_ENDPOINT_ID);
+                        ArrayList<String> args = this.getIntent().getStringArrayListExtra(SESSION_ENDPOINT_ARGUMENTS);
```

Comment:
Wouldn't these arguments need to be a Map (name to value) rather than a list?

Reply:
^ If this were true, we might also want to consider having the args provided as a bundle map.